### PR TITLE
Feat: add new design for categories page 

### LINKED ISSUE #2177: feat: add new design for categories page
> You can view the figma file [here](https://www.figma.com/file/ntisIxc9QHwOZ828CH5G3a/LinksHub?type=design&node-id=17-231&mode=design&t=29U3Qw0LktnVRteR-0).

image

> The design should be responsive. 

---

ℹ️ We recommend updating the styling of existing components, otherwise you can add extra components if there is a need.

- You also need to implement breadcrumbs based on the current category we are currently on.

image

> Any changes to the above design are welcome. Please feel free to share your thoughts, and we will discuss them thoroughly.

---

### PR DESCRIPTION
**Pull request by gakshita**
fixes #2177 

## Changes proposed
New design for category page implemented

### CODE REVIEW

**Review by CBID2 — COMMENTED** [review]
The page looks good @gakshita, but can you revert [the commits that don't pertain to the issue that your PR solves?](https://github.com/rupali-codes/LinksHub/pull/2195/commits)

**Comment by gakshita** [comment]
> The page looks good @gakshita, but can you revert [the commits that don't pertain to the issue that your PR solves?](https://github.com/rupali-codes/LinksHub/pull/2195/commits)

Have removed the unecessary commits

File: assets/icons/AddIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { SVGProps } from 'react'
+const AddIcon = ({ ...rest }: SVGProps<SVGSVGElement>) => {
```
Comment (aftabrehan) [inline assets/icons/AddIcon.tsx:3]: I don't think it's a better approach to create a simple SVG icon as a component. Can you please create `.svg` file and just simply import it? It'll work as a React component and will accept props (e.g. `className`).

File: pages/[category]/index.tsx [hunk 2]
```diff
@@ -1,56 +1,91 @@
...
-                {category.split('-').join(' ')}
-              </p>
-              <p className="text-xl hidden md:block">
-                Get access to all exclusive{' '}
-                <span className="capitalize text-theme-primary">
-                  {category.split('-').join(' ')}
-                </span>{' '}
-                resources!
-              </p>
+            <div className="flex mt-8 justify-between	">
```
Comment (aftabrehan) [inline pages/[category]/index.tsx:23]: Please, avoid unnecessary spaces.

```suggestion
            <div className="flex mt-8 justify-between">
```

File: pages/[category]/index.tsx [hunk 3]
```diff
@@ -1,56 +1,91 @@
...
+              <BreadCrumbs
+                sections={[{ name: category, url: `/${category}` }]}
+              />
+              <div className="flex my-auto gap-x-3">
+                <MessageIcon />
+                <div className="text-md text-text-quinary hidden md:block">
```
Comment (aftabrehan) [inline pages/[category]/index.tsx:29]: Please, use a `span` tag if you don't need to render it in a new line.

```suggestion
                <span className="text-md text-text-quinary hidden md:block">
```

File: pages/[category]/index.tsx [hunk 4]
```diff
@@ -1,56 +1,91 @@
...
-            <div className="flex flex-wrap gap-3 mt-6 justify-center items-start mx-auto lg:max-w-sm max-md:max-w-sm">
-              {subcategories.map((subcat, i) => (
+            <div className="flex max-w-1038 justify-center">
+              <div className="flex flex-wrap gap-x-9 gap-y-5 mt-6 mb-28 flex-start items-start">
+                {subcategories.map((subcat, i) => (
+                  <Link
+                    key={i}
+                    href={`/${category}${subcat.url}`}
+                    aria-label={`Explore ${subcat.name}`}
+                    className={`group flex border dark:border-theme-primary dark:border-opacity-8 rounded-xl bg-theme-tertiary  hover:bg-theme-quaternary hover:bg-opacity-25	 px-3 py-1 dark:hover:border-opacity-25 hover:text-dark-primary shadow-lg lg:max-w-xs max-md:max-w-xs px-4 py-4 group`}
```
Comment (aftabrehan) [inline pages/[category]/index.tsx:41]: Any specific reason for using template literals here?

File: pages/[category]/index.tsx [hunk 5]
```diff
@@ -1,56 +1,91 @@
...
+                  >
+                    <div>
+                      <div
+                        className={`dark:text-text-primary text-lg capitalize font-semibold	`}
```
Comment (aftabrehan) [inline pages/[category]/index.tsx:45]: Please, use template literals only when it's required and avoid extra spaces.

```suggestion
                        className="dark:text-text-primary text-lg capitalize font-semibold"
```

File: pages/[category]/index.tsx [hunk 6]
```diff
@@ -1,56 +1,91 @@
...
+                      >
+                        {' '}
```
Comment (aftabrehan) [inline pages/[category]/index.tsx:47]: Do we really require this space, or it is added prettier when saving the file?

File: components/Breadcrumps/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    <>
+      <div className="hidden md:flex">
+        <Link
+          href={`/`}
+          className="flex gap-x-2 bg-theme-quinary bg-opacity-20 text-text-quaternary text-lg py-0.5 px-2.5 rounded-lg cursor-pointer	"
+        >
+          <HomeIcon />
+          <div>Home</div>
+        </Link>
+        <div className="text-2xl	mx-2.5 font-semibold varela">/</div>
```
Comment (aftabrehan) [inline components/Breadcrumps/index.tsx:19]: Any specific reason for using `varela` font?

**Review by aftabrehan — CHANGES_REQUESTED** [review]
Excellent PR, @gakshita 💪 🚀 💯 

Thanks for the contribution. It just needs minor adjustments and then we're good to go. 

However, I have only reviewed the code yet. I'll proceed with UI testing and provide my feedback in a while. Feel free to ask if you have any questions or want clarification on something.

**Comment by gakshita** [comment]
@aftabrehan @rupali-codes @CBID2 all changes have been fixed from my side!

**Review by aftabrehan — COMMENTED** [review]
Thanks @gakshita! The code looks much better now.

However, here are a few suggestions from my side.

**Comment by aftabrehan** [comment]
It all looks super awesome 🙌. Great work @gakshita. Please, help to adjust these two things as well, and then we're good to merge this PR.

1. Can you please add the background on the parent element?
<div align="center">
Screenshot 2023-11-30 at 6 06 59 PM
</div>

3. Can you please use a full width for the tile on the mobile screen to cover extra space?
<div align="center">
Screenshot 2023-11-30 at 6 14 45 PM
</div>

File: pages/[category]/index.tsx [hunk 8]
```diff
@@ -38,39 +45,33 @@ const CategoryPage = () => {
...
-                      >
-                        {' '}
+                      <div className="dark:text-text-primary text-text-primary-light text-lg capitalize font-medium">
                         {subcat.name}
                       </div>
-                      <div
-                        className={`dark:text-text-tertiary text-md font-medium w-10/12 h-12	overflow-y-hidden`}
-                      >
+                      <div className="dark:text-text-tertiary text-text-quinary text-md font-medium w-10/12 h-12 overflow-y-hidden">
                         Explore {subcat.name} resources for learning and growth.
```
[the remark this answers is not in the document]
Comment (CBID2) [reply pages/[category]/index.tsx:55]: Good point @aftabrehan

**Comment by Anmol-Baranwal** [comment]
@gakshita 
Are you facing any problems regarding the changes? Please let me know; I will gladly help you.

**Comment by gakshita** [comment]
Couldn't find item description data i the object, rest changes have been pushed.

**Comment by CBID2** [comment]
Any updates @gakshita? 😁

**Comment by rupali-codes** [comment]
@gakshita you still working on it?

**Pull request merged**